sonic-xcvrd/xcvrd/xcvrd.py

```diff
@@ -1150,6 +1182,14 @@ def task_worker(self):
                              self.CMIS_STATE_REMOVED]:
                     continue
 
+                # Handle the case when Xcvrd was NOT running when 'host_tx_ready' or 'admin_status'
+                # was updated or this is the first run so reconcile the above two attributes
+                if 'host_tx_ready' not in self.port_dict[lport]:
```

Comment:
This code will not get hit if the port is CMIS_STATE_READY, so with config-interface shutdown the event will be returned in the above if condition and the modules will not be put in DP-DEACTIVATE. 

If any runtime Tx fault handler being implemented and the host_tx_ready is set to false, then the event will be missed.

Reply:
CMIS state reset happens if any port configuration is changed including admin or host_tx_ready status here :[https://github.com/Azure/sonic-platform-daemons/blob/master/sonic-xcvrd/xcvrd/xcvrd.py#L919](https://github.com/Azure/sonic-platform-daemons/blob/master/sonic-xcvrd/xcvrd/xcvrd.py#L919)